engine/src/test/java/io/camunda/zeebe/engine/processing/job/MultiTenancyActivatableJobsPushTest.java

```diff
@@ -0,0 +1,139 @@
+import io.camunda.zeebe.protocol.record.intent.JobBatchIntent;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.JobBatchRecordValue;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import io.camunda.zeebe.test.util.Strings;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.agrona.DirectBuffer;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MultiTenancyActivatableJobsPushTest {
+
+  private static final String PROCESS_ID = "process";
+
+  private static final RecordingJobStreamer JOB_STREAMER = new RecordingJobStreamer();
+
+  @ClassRule
+  public static final EngineRule ENGINE =
+      EngineRule.singlePartition().withJobStreamer(JOB_STREAMER);
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  private final List<Long> activeProcessInstances = new ArrayList<>();
+
+  @Test
+  public void shouldPushWhenJobCreatedForAuthorizedTenant() {
+    // given
+    final String jobType = Strings.newRandomValidBpmnId();
+    final DirectBuffer jobTypeBuffer = BufferUtil.wrapString(jobType);
+    final DirectBuffer worker = BufferUtil.wrapString("test");
+    final Map<String, Object> variables = Map.of("a", "valA", "b", "valB", "c", "valC");
+    final long timeout = 30000L;
+    final String tenantIdA = "tenant-a";
+    final String tenantIdB = "tenant-a";
+
+    final JobActivationPropertiesImpl jobActivationProperties =
+        new JobActivationPropertiesImpl()
+            .setWorker(worker, 0, worker.capacity())
+            .setTimeout(timeout)
+            .setFetchVariables(
+                List.of(new StringValue("a"), new StringValue("b"), new StringValue("c")));
+    final var jobStreamA =
+        JOB_STREAMER.addJobStream(
+            jobTypeBuffer, jobActivationProperties.setTenantIds(List.of(tenantIdA)));
+    final var jobStreamB =
+        JOB_STREAMER.addJobStream(
+            jobTypeBuffer, jobActivationProperties.setTenantIds(List.of(tenantIdB)));
+
+    final int activationCount = 1;
+
+    // when
+    final long jobKey = createJob(jobType, PROCESS_ID, variables, tenantIdA);
+
+    // then
+    final Record<JobBatchRecordValue> batchRecord =
+        jobBatchRecords(JobBatchIntent.ACTIVATED).withType(jobType).getFirst();
+
+    // assert job batch record
+    final JobBatchRecordValue batch = batchRecord.getValue();
+    final List<JobRecordValue> jobs = batch.getJobs();
+    assertThat(jobs).hasSize(1);
+    assertThat(batch.getJobKeys()).contains(jobKey);
+
+    // assert event order
+    assertEventOrder(JobIntent.CREATED, JobBatchIntent.ACTIVATED);
+
+    // assert job stream
+    assertActivatedJob(jobStreamA, jobKey, worker, variables, activationCount, tenantIdA);
```

Comment:
❌ For completeness sake we should also assert that the job was not pushed to jobstreamB

Reply:
I've added this myself in https://github.com/camunda/zeebe/pull/14464/commits/12f03e7c8e9727b789e21545fedb1c9cb2c81bfb